### PR TITLE
Helm values generator

### PR DESCRIPTION
**Pull request by ozdanborne**
## Description

There's two commits in this PR, I'd recommend viewing each commit first to understand.

The first commit is a cleanup of the ruby processing. We were loading `_config.yml` from disk when we didn't have to - jekyll already does that and loads it into `context.registers[:site]`. The first commit uses that value instead.

The second commit moves the values.yaml generation into a separate ruby file, then attaches a CLI frontend onto it. This will allow us to quickly generate a valid values.yaml file for helm without having to duplicate or move around any of the information stored in `_data/versions.yml`. 

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 0486c8380127b0380e99866b447f8943c9348764

https://deploy-preview-2492--calico.netlify.com

File: hack/gen_values_yml.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+usage = "ruby hack/gen_values_yaml.rb <version> [arguments...]
+
+It's recommended to run this from the root of the Calico repository,
+as the default paths assume as much.
+
+<version> should be the major.minor version (e.g. v3.6) or master.
+
+--config    Path to the jekyll config. [default: _config.yml]
+--versions  Path to the versions.yml. [default: _data/versions.yml]
```
Comment (caseydavenport) [inline hack/gen_values_yml.rb]: Do we need more than just the versions yaml?

Do we want the ability to pick _which_ version to generate?

e.g. `--version master` versus `--version v3.6.0` ?

File: hack/gen_values_yml.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
+require "optparse"
+require "yaml"
+require_relative "../_plugins/lib"
+
+usage = "ruby hack/gen_values_yaml.rb <version> [arguments...]
```
Comment (caseydavenport) [inline hack/gen_values_yml.rb]: Ignore my other comment, I can't read.

**Comment by caseydavenport** [comment]
@ozdanborne this looks nice.

WDYT about wrapping this script in a make target that runs it within a docker image so we remove the dependency on ruby?

**Comment by ozdanborne** [comment]
@caseydavenport good idea, will do.

I'm going to close this and pull it into #2433, as the two are closely related.

**Pull request closed without merging**